### PR TITLE
Standardize attendances

### PR DESCRIPTION
**Pull request by RobertAKARobin**
Also moves attendance/submission to own partials for extra dryness.

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -85,15 +85,6 @@ def markdown(text)
...
-  def status_button record, status
-    output = ""
-    id = "#{dom_id(record)}_#{status}"
-    checked = "checked" if record.status == status
-    output += "<input type='radio' name='#{dom_id(record)}' id='#{id}' value='#{status}' #{checked} data-record-url='#{url_for record}' data-record-attribute='status' />"
-    output += "<label for='#{id}' class='status'>#{status}</label>"
-    return output.html_safe
-  end
-
   def cohort_status(cohort, user)
```
Comment (andrewsunglaekim) [inline app/helpers/application_helper.rb]: I've always been confused whether to use helpers vs partials. Is there any kind of rule or convention?

Comment (RobertAKARobin) [inline app/helpers/application_helper.rb]: Yeah, I dunno. After looking back at this, it seems to me that if something involves writing a bunch of HTML then it's better as a partial.

File: app/views/components/_todo_table.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<% class_name   = collection.model_name %>
+<%= render partial: "#{class_name.plural}/thead" %>
+<% collection.each do |record| %>
+  <%= render partial: "#{class_name.plural}/tr", locals: {
+    :"#{class_name.singular}"   => record,
```
Comment (andrewsunglaekim) [inline app/views/components/_todo_table.html.erb]: whoa, this is really cool, i had no idea you could make dynamic partials like this... this is sick.

File: app/views/submissions/_tr.html.erb [hunk 3]
```diff
@@ -18,32 +21,22 @@
...
-            <% id = "#{submission.id}-status-#{status}" %>
-            <%= f.radio_button :status, status, id: id %>
-            <%= f.label :status, status, for: id %>
-          <% end %>
-        </div>
+        <%= f.text_field :score, class: "sm", id: nil %><%= out_of_score %>
       <% end %>
     <% else %>
-      <%= submission.status %>
+      <%= "#{submission.score}#{out_of_score}" %>
```
Comment (andrewsunglaekim) [inline app/views/submissions/_tr.html.erb]: this feels like it should be a model method for submissions

Comment (RobertAKARobin) [inline app/views/submissions/_tr.html.erb]: Could be... I only envision it showing up in HTML, though.

File: app/views/components/_status_buttons.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<% klazz = record.model_name.to_s.classify.constantize %>
+<%= form_for record, html:{id: nil}, data:{autosave: true} do |f| %>
```
Comment (andrewsunglaekim) [inline app/views/components/_status_buttons.html.erb]: el o el

**Comment by andrewsunglaekim** [comment]
this is a really cool PR, i don't know enough about conventions for these things for my thumbs up to hold much value but :+1: from me. My comments are more just cause i'm curious

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (andrewsunglaekim) [inline app/helpers/application_helper.rb]: yeah, taht sounds right

**Comment by RobertAKARobin** [comment]
Thanks, man! I really appreciate it!

**Comment by amaseda** [comment]
:+1:

**Pull request merged**